nio/client/async_client.py

```diff
@@ -3431,3 +3435,52 @@ async def room_upgrade(self,
                 return RoomUpgradeError("Could update the new rooms aliases")
 
         return RoomUpgradeResponse(new_room.room_id)
+
+    @logged_in
+    async def has_event_permission(self,
+                                   room_id: str,
+                                   event_name: str,
+                                   event_type: str = "event") -> Union[bool, ErrorResponse]:
+        who_am_i = await self.whoami()
+        power_levels = await self.room_get_state_event(room_id, "m.room.power_levels")
+
+        try:
+            user_power_level = power_levels.content['users'][who_am_i.user_id]
+        except KeyError:
+            user_power_level = power_levels.content['users_default']
+        else:
+            return ErrorResponse("Couldn't get user power levels")
+
+        try:
+            event_power_level = power_levels.content['events'][event_name]
+        except KeyError:
+            if event_type == "event":
+                event_power_level = power_levels.content['events_default']
+            elif event_type == "state":
+                event_power_level = power_levels.content['state_default']
+            else:
+                return ErrorResponse("event_type {} unknown".format(event_type))
```

Comment:
if we replace the `event_type` string with a `state` boolean, then we don't have to handle that case here.